cmds/core/wget/wget_test.go

```diff
@@ -98,25 +99,31 @@ var tests = []struct {
 	},
 }
 
-func getFreePort(t *testing.T) int {
+func getListener(t *testing.T) (net.Listener, int) {
+	t.Helper()
 	l, err := net.Listen("tcp", ":0")
 	if err != nil {
-		t.Fatalf("Cannot create free port: %v", err)
+		t.Fatalf("error setting up TCP listener: %v", err)
 	}
-	l.Close()
-	return l.Addr().(*net.TCPAddr).Port
+	return l, l.Addr().(*net.TCPAddr).Port
 }
 
 // TestWget implements a table-driven test.
 func TestWget(t *testing.T) {
 	// Start a webserver on a free port.
-	unusedPort := getFreePort(t)
-
-	l, err := net.Listen("tcp", ":0")
-	if err != nil {
-		t.Fatalf("Cannot create free port: %v", err)
-	}
-	port := l.Addr().(*net.TCPAddr).Port
+	l, port := getListener(t)
+	ul, unusedPort := getListener(t)
+	defer ul.Close()
```

Comment:
I think you also need a "defer l.Close()", correct?

Reply:
Updated. It needs to change one line below otherwise the test will fail for terminating the connection.